File: java/src/test/java/com/restfulbooker/api/payloads/request/AuthPayload.java
package com.restfulbooker.api.payloads.request;

import org.codehaus.jackson.annotate.JsonProperty;

public class AuthPayload {

    @JsonProperty
    private String username;
    @JsonProperty
    private String password;

    private AuthPayload(String username, String password) {
        this.username = username;
        this.password = password;
    }

    public String getUsername() {
        return username;
    }

    public String getPassword() {
        return password;
    }

    public static class AuthPayloadBuilder{

        private String username;
        private String password;

        public AuthPayloadBuilder setUsername(String username) {
            this.username = username;
            return this;
        }

        public AuthPayloadBuilder setPassword(String password) {
            this.password = password;
            return this;
        }

        public AuthPayload build(){
            return new AuthPayload(username, password);
        }
    }
}

File: csharp/Payloads/AuthPayload.cs
namespace Payloads
{
    public class AuthPayload
    {
        public string Username { get; private set; }
        public string Password { get; private set; }

        private AuthPayload(string username, string password)
        {
            Username = username;
            Password = password;
        }

        public AuthPayload()
        {
        }


        public void SetUsername(string username)
        {
            Username = username;
        }

        public void SetPassword(string password)
        {
            Password = password;
        }

    }
}
